client/tx/tx.go

```diff
@@ -256,7 +256,7 @@ func BuildSimTx(txf Factory, msgs ...sdk.Msg) ([]byte, error) {
 	// sentinel pubkey.
 	sig := signing.SignatureV2{
 		Data: &signing.SingleSignatureData{
-			SignMode: txf.signMode,
+			SignMode: txf.WithSignMode(signing.SignMode_SIGN_MODE_LEGACY_AMINO_JSON).signMode,
```

Comment:
I feel like this is suspicious to set SIGN_MODE_LEGACY_AMINO. I thought simulation was switched over to SIGN_MODE_DIRECT

Reply:
This is specifically for the gas simulation. I don't think it makes a huge difference but not 100% sure

Reply:
Yeah, signatures aren't verified in simulation mode, so I believe it doesn't make a difference.

Anyways, it's reverted here: https://github.com/cosmos/cosmos-sdk/pull/7207/files#diff-df9cc9b2c6d08e86dbf0554ea3adb00aR259

Reply:
This was causing a test to fail but maybe we fixed it another way 🤷‍♂️  fine with reverting